impl/pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,86 @@
+package telemetry
+
+import (
+	"context"
+	"runtime/debug"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"go.opentelemetry.io/contrib/instrumentation/runtime"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const scopeName = "github.com/TBD54566975/did-dht-method"
+
+var (
+	Tracer        trace.Tracer
+	traceProvider *sdktrace.TracerProvider
+
+	Meter         metric.Meter
+	meterProvider *sdkmetric.MeterProvider
+
+	version = "unversioned"
+)
+
+func init() {
+	buildInfo, ok := debug.ReadBuildInfo()
```

Comment:
interesting...how does this work?

we do/should have a version set in our config

Reply:
Read about it [here](https://pkg.go.dev/runtime/debug#ReadBuildInfo), right now it's just returning `(devel)` for the current module's version:

![Screenshot 2024-03-05 at 10 27 05 AM](https://github.com/TBD54566975/did-dht-method/assets/139806439/dc8b0797-e644-41b6-b724-1961c0cb9ce9)

I thought that would become a useful name when we add tags, but I just tried adding a tag and it didn't help, so I might try to push that forward a little before merging this.

Reply:
Okay so i thought i had successfully used this in the past but i guess [it will never work](https://github.com/golang/go/issues/29228). Looking at where I thought I had successfully used it, I appear to have some compile-time variable setting to inject a version externally. Will add the same here.